k8s/k8s.go

```diff
@@ -91,19 +93,48 @@ func (k8s *Client) PodByIP(IP string) (*v1.Pod, error) {
 	}
 
 	if len(pods) == 0 {
+		metrics.PodNotFoundInCache.Inc()
 		return nil, fmt.Errorf("pod with specificed IP not found")
 	}
 
 	if len(pods) == 1 {
 		return pods[0].(*v1.Pod), nil
 	}
 
-	//This happens with `hostNetwork: true` pods
-	podNames := make([]string, len(pods))
-	for i, pod := range pods {
-		podNames[i] = pod.(*v1.Pod).ObjectMeta.Name
+	if !k8s.dealWithDupIP {
```

Comment:
I would suggest swapping the order here:

```
if k8s.dealWithDupIP {
  // Handle
}
// Else handle default case
```

Reply:
I prefer this way better so that the diff stays minimal.
Not addressing this suggestion.